File: 0136.py
from typing import List

class Solution:
  def singleNumber(self, nums: List[int]) -> int:
    res = 0
    for n in nums:
      res = n ^ res
    return res

solution = Solution()

# case 1
nums = [2,2,1]
result = solution.singleNumber(nums)
print(result)

# case 2
nums = [4,1,2,1,2]
result = solution.singleNumber(nums)
print(result)

# case 3
nums = [1]
result = solution.singleNumber(nums)
print(result)

File: 0136.ts
function singleNumber(nums: number[]): number {
  let res = 0;
  for (const n of nums) {
    res = res ^ n;
  }
  return res;
}

export {
  singleNumber
}

// case 1
let nums: number[] = [2,2,1];
let result: number = singleNumber(nums);
console.log(result);

// case 2
nums = [4,1,2,1,2];
result = singleNumber(nums);
console.log(result);

// case 3
nums = [1];
result = singleNumber(nums);
console.log(result);